src/repository/models.py

```diff
@@ -0,0 +1,1033 @@
+    )
+    abstract = models.TextField(
+        blank=True,
+        null=True,
+        help_text=_(
+            'Please avoid pasting content from word processors as they can add '
+            'unwanted styling to the abstract. You can retype the abstract '
+            'here or copy and paste it into notepad/a plain text editor before '
+            'pasting here.',
+        )
+
+    )
+
+    def approve(self):
+        self.date_decision = timezone.now()
+        self.approved = True
+        current_version = None
+
+        # Update the current version to have the Preprint's current title
+        # and abstract.
+        if self.preprint.current_version is not None:
+            current_version = self.preprint.current_version
+            current_version.title = self.preprint.title
+            current_version.abstract = self.preprint.abstract
+            current_version.published_doi = self.preprint.doi
+            this_file = self.preprint.current_version.file
+        # no version yet
+        else:
+            this_file = self.preprint.submission_file
+
+        # Create a new PreprintVersion, this will now be the current_version.
+        # If the current VersionQueue has no file (in the case of Metadata
+        # updates) use the preprint's current version's file.
+        PreprintVersion.objects.create(
+            preprint=self.preprint,
+            file=self.file if self.file else this_file,
+            version=self.preprint.next_version_number(),
+            moderated_version=self,
+            published_doi=self.published_doi,
+        )
+
+        # Overwrite the preprint's metadata now we have a historical record.
+        # Check that title and abstract have value, if not there is no change.
+        if self.title:
+            self.preprint.title = self.title
+        if self.abstract:
+            self.preprint.abstract = self.abstract
+        if self.published_doi:
+            self.preprint.doi = self.published_doi
+
+        if current_version is not None:
+            current_version.save()
+        self.preprint.save()
+        self.save()
+
+    def decline(self):
+        self.date_decision = timezone.now()
+        self.approved = False
+        self.save()
+
+    def decision(self):
+        if self.date_decision and self.approved:
+            return True
+        elif self.date_decision:
+            return False
+
+    def status(self):
+        if self.date_decision and self.approved:
+            return _('Approved')
+        elif not self.date_decision:
+            return _('Under Review')
+        else:
+            return _('Declined')
+
+
+@receiver(models.signals.post_delete, sender=PreprintFile)
+def auto_delete_file_on_delete(sender, instance, **kwargs):
+    """
+    Deletes file from filesystem
+    when corresponding `PreprintFile` object is deleted.
+    """
+    if instance.file:
+        if os.path.isfile(instance.file.path):
+            os.remove(instance.file.path)
```

Comment:
Is this desirable? This is not what we do when deleting on the journal side, we always keep the files.

In any case, it should probably catch the exception in case the file does not exist.